### PR TITLE
Render <img> with correct index and a reference to the list of links

### LINKED ISSUE #1506: Rendering of <img> is inconsistent with other media types
**Newsboat version**: commit c783cb454622e75b1d9bb1bf6401bcb7cd37618c

Config file: n/a

I would expect Newsboat to render, for example, the first image placeholder in an article, which appears after a single `<a>`, as `[image 1 (link #2)]` rather than `[image 2]`, similar to how `<audio>`, `<video>`, etc. are rendered. In general, I would expect all image placeholders to display an index between 1 and `image_count` (inclusive), but currently it [displays the `linknum`](https://github.com/newsboat/newsboat/blob/c783cb454622e75b1d9bb1bf6401bcb7cd37618c/src/htmlrenderer.cpp#L341) instead (which will fall outside that range if there are any other links before any images). Since the current behavior is inconsistent with other media types, I believe it is confusing.

Steps to reproduce the issue:

1. Checkout the branch `bugfix/img-numbering` from https://github.com/markmatney/newsboat.git

2. Run `TMPDIR=/dev/shm make -j5 PROFILE=1 check` and observe the test that fails

### PR DESCRIPTION
**Pull request by markmatney**
Fixes #1506.

### CODE REVIEW

File: src/htmlrenderer.cpp [hunk 1]
```diff
@@ -304,49 +304,27 @@ void HtmlRenderer::render(std::istream& input,
...
-					} else {
-						curline.append(strprintf::fmt(
-								"[%s %u]",
-								_("image"),
-								link_num));
-					}
-					image_count++;
+				if (img_url.length() > 0) {
+					add_media_link(curline, links, url, img_url,
+						img_title, image_count, LinkType::IMG);
```
Comment (Minoru) [inline src/htmlrenderer.cpp:327]: `add_media_link()` does nothing if the fourth argument is empty, so this `if` can be removed.

File: src/htmlrenderer.cpp [hunk 2]
```diff
@@ -1030,22 +1008,29 @@ void HtmlRenderer::add_media_link(std::string& curline,
...
 	if (!media_title.empty()) {
-		output = strprintf::fmt("[%s %u: %s (%s #%u)]", _(type_str.c_str()),
-				media_count, media_title, _("link"), link_num);
+		output = strprintf::fmt("[%s %u: %s (link #%u)]", _(type_str.c_str()),
+				media_count, media_title, link_num);
 	} else {
-		output = strprintf::fmt("[%s %u (%s #%u)]", _(type_str.c_str()),
-				media_count, _("link"), link_num);
+		output = strprintf::fmt("[%s %u (link #%u)]", _(type_str.c_str()),
+				media_count, link_num);
```
Comment (Minoru) [inline src/htmlrenderer.cpp:1033]: `_()` is a macro that expands to `gettext()` (see config.h in the root of the repo). We use this to internationalize the application. This change makes "link" untranslatable, so please revert this.

File: src/htmlrenderer.cpp [hunk 3]
```diff
@@ -1030,22 +1008,29 @@ void HtmlRenderer::add_media_link(std::string& curline,
 	const std::string& media_url, const std::string& media_title,
 	unsigned int media_count, LinkType type)
 {
+	std::string link_url;
```
Comment (Minoru) [inline src/htmlrenderer.cpp:1011]: It's better to declare variables close to their use-site, to make the code slightly easier to understand ;)

File: src/htmlrenderer.cpp [hunk 4]
```diff
@@ -1030,22 +1008,29 @@ void HtmlRenderer::add_media_link(std::string& curline,
...
+		link_url = "inline image";
+	} else {
+		link_url = utils::censor_url(utils::absolute_url(url, media_url));
+	}
+
 	std::string type_str = type2str(type);
-	unsigned int link_num = add_link(links,
-			utils::censor_url(utils::absolute_url(url, media_url)),
-			type);
+	unsigned int link_num = add_link(links, link_url, type);
```
Comment (Minoru) [inline src/htmlrenderer.cpp:1025]: Since you're editing the line anyway, you could make this `const` ;)

**Review by Minoru — APPROVED** [review]
Thank you! Since it's your second PR, it's time to get nitpicky and tell you about some of the newer code conventions we have :) Not all code follows them, but we try to upgrade it as we go. The only thing that *must* be changed is the call to `_()`; the rest of my comments are just nitpicks.

**Comment by Minoru** [comment]
Something's up with the code coverage job, too. I don't think it's because of your PR, but I'll have to wait for that job to fail so I can see the logs and figure out what's going on (hopefully).

File: src/htmlrenderer.cpp (hunk 2 above)
Comment (markmatney) [reply src/htmlrenderer.cpp:1033]: ah, thanks for explaining that. I admit I did look at config.h to try to figure out what `_()` does, but with my C++ being rather rusty (and having never properly learned about macros) ... well ... this happened :sweat_smile: I should have run `man gettext`

**Review by Minoru — APPROVED** [review]
Excellent, thank you!

I'll merge this once I figured out what's going on with that Coveralls CI job.

**Pull request merged**